src/backend/MaGe/utils.ts

```diff
@@ -1,29 +1,36 @@
 // ------------------ MATH & STATISTICS ------------------
 
-// calculates the cartesian product from two array
-function cartesianProduct<T>(...allEntries: T[][]): T[][] {
-    return allEntries.reduce<T[][]>(
-      (results, entries) =>
-        results
-          .map(result => entries.map(entry => result.concat([entry])))
-          .reduce((subResults, result) => subResults.concat(result), []),
-      [[]]
-    );
-}
-
+/** Coding magic that I have no idea what it does :)
+ * To me, this is little piece of code is coding magic as I really have no idea how this calculates the cartesian product.
+ * The editor cannot even infer what types are used
+ * But, given that it works, I would say that maybe it is science so advanced that it is considered as magic :P
+ * @param a an array
+ * @param b an array
+ * @returns an array of arrays
+ */
 const f = (a, b) => [].concat(...a.map(d => b.map(e => [].concat(d, e))));
+/** A function that calculates the cartesian of arrays using the above-mentioned coding magic :D
```

Comment:
not super supportive of these style of code but I stumble upon this rather often so it can be kept.